Add content/en/hypervisor (#1385)
cncf/glossary: content/en/hypervisor.md

---
title: Hypervisor
status: Feedback Appreciated
category: Technology
tags: ["application", "", ""]
---

## What it is

> A hypervisor is a software application that performs [virtualization](/virtualization/)

Reviewer: @KiranSatyaRaj 
I think `software application` may not correct technically.
In general, there are two types of hypervisor. 
 1) bare metal/native hypervisor that runs on bare metal computer directly (e.g., KVM, Xen, ...)
 2) hosted hypervisor that runs on OS as an application (e.g., Virtual box)

So, if we say `A hypervisor is a software application`, readers may be confused with the second type of hypervisor.

---
title: Hypervisor
status: Feedback Appreciated
category: Technology
tags: ["application", "", ""]
---

## What it is

The Hypervisor is a software application that performs [virtualization](/virtualization/)

> by taking advantage of [bare metal machine](/bare-metal-machine/) resources

Reviewer 2: ```suggestion
by taking the advantage of [bare metal machine](/bare-metal-machine/) resources
```

(CPU, Memory, Network, and Storage) , dividing them into sub-parts, 
and allocating resources according to business logic to create [virtual machines (VM)](/virtual-machine/) and the ability to run multiple VMs simultaneously.

> In the marketplace, they are available as Type 1 hypervisors that can be installed directly on the physical server, 

Author: yeah, you're right, knowing its functionality is more important than types so I'll add some more information and I'll get back to you with that.

giving it direct access to system hardware resources.
Type 2 hypervisors, on the other hand, are installed and 
executed by the operating system and allow the operating system to make hardware decisions.
## Problem it addresses

Traditionally, a server could only run one operating system at a time.

> Devoting all the resources of the host machine to a single service was not the most effective way.

Reviewer 3: ```suggestion
Devoting all the resources of that server to just a single application was not the most effective way.
```

> Considering the capabilities of a server it can run multiple applications.

Reviewer 3: The content of the rest of this section is not very accurate, or easy to follow. The aspect of having wasted a whole powerful server on just one app is an important aspect that can be followed up on. For example just saying, that a server could hold more apps, but then you have isolation and noisy neighbors problems (one highly resource-consuming app has an impact on the other ones). 
Another problem of the world before virtualization is setup costs. VMs are more quickly set up than bare metal. This is one step to closing the gap between dev and ops.

Author: I'm thinking these points could go into this section " acquiring a physical machine takes time, it requires a team of engineers to monitor them, whereas VMs are more quickly set up than bare metal", what do you think? @iamNoah1 and is it because a hypervisor takes away the low-level complexities (like abstraction layer over the physical hardware) so that it is easier to set up a VM whether they are of the dev team or Ops team?

There is a need for software that can provide the same environment as a physical machine to the applications within it.
Multiply the environments and commit resources to them as needed, ensuring their security simultaneously.
So that the data of one cannot be accessed by the other.

## How it helps

> The Hypervisor is used to create and allocate resources to virtual machines until we reach the saturating performance of the physical machines,

Reviewer 3: ```suggestion
The Hypervisor is used to create and allocate resources to virtual machines until we reach the saturating performance of the physical machines.
```

> conserving resources and reducing energy consumption.

Reviewer 3: ```suggestion

```

> A hypervisor simplifies the migration in its current state in the event of a failure on the host server.

Reviewer 3: ```suggestion
```

as it does not really add value to the definition

> The reason for this is that it maintains snapshots of the workload so that it can be transported from one machine to another.

Reviewer 3: ```suggestion

```

In the context of [cloud computing](/cloud-computing/), the hypervisor becomes an effective tool.

> Organizations can migrate their applications to the cloud to take advantage of [cloud native apps](/cloud-native-apps/) and autoscale their services.

Reviewer 3: I would try to focus more on the setup speed and the degree of isolation between apps than on migrating for the rest of this section.
